spec/controllers/attributes_controller_spec.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require Rails.root.join('spec', 'support', 'mock_auth_helper.rb')
+
+describe AttributesController do
```

Comment:
Again, I don't approve of controller tests. I think you should be writing integration tests: http://guides.rubyonrails.org/testing.html#integration-testing